### PR TITLE
Add TypeScript declarations?

### PR DESCRIPTION
**Pull request by bclinkinbeard**
I'm not sure if you're open to this, but I think it would be helpful to anyone using TS. I'd recommend a second set of eyes before merging, but I think this file is all that's needed.

### CODE REVIEW

**Comment by bclinkinbeard** [comment]
Should I take this to mean you'd rather keep the defs in DefinitelyTyped? :)

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export as namespace KDBush
+
+/*~ This declaration specifies that the class constructor function
+ *~ is the exported object from the file
+ */
+export = KDBush
+
+declare class KDBush {
+  constructor(
+    points: [number, number][],
```
Comment (vicapow) [inline index.d.ts]: points doesn't have to be a tuple

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    getX?: (p: any) => number,
```
Comment (vicapow) [inline index.d.ts]: `any` is a weak type. can you use a type generics in TS?

**Comment by mourner** [comment]
Sorry for getting around to this so late! I took a different route in https://github.com/mourner/kdbush/commit/d1d856fb12f52de642c3ac721bf7e35cd63e3478

**Pull request closed without merging**